teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

```diff
@@ -0,0 +1,239 @@
+---
+status: proposed
+title: Allow custom task to be embedded in pipeline
+creation-date: '2021-03-18'
+last-updated: '2021-03-27'
+authors:
+- '@Tomcli'
+- '@litong01'
+- '@ScrapCodes'
+---
+
+# TEP-0061: Allow custom task to be embedded in pipeline
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Tektoncd/Pipeline currently allows custom task to be referenced in pipeline
+resource specification file using [`taskRef`](https://github.com/tektoncd/community/blob/main/teps/0002-custom-tasks.md).
+That feature allows a custom task to be submitted to kubernetes along
+the submission of the [Tektoncd/pipeline](https://github.com/tektoncd/pipeline),
+however, the submission of a custom task is a separate request to Kubernetes.
```

Comment:
Could you clarify this a bit?  It sounds in the beginning like custom tasks are submitted to k8s along with a pipeline but then the second part of the sentence says  that, no, custom tasks are actually submitted separately.  I don't totally understand what we mean here.

Reply:
Sorry, there are some mistakes on the wording. The formal custom task is referring to the pipelineTask type. The latter custom task is referring to the custom resources created for the custom task. I will update the summary to make it more clear.